source/mod_dist.f90

```diff
@@ -98,159 +265,726 @@ subroutine dist_parseInputLine(inLine, iLine, iErr)
+  case("ION_Z")
+    call dist_appendFormat(dist_fmtIonZ,     one,  0)
+  case("PDGID")
+    call dist_appendFormat(dist_fmtPDGID,    one,  0)
+
+  case("4D") ! 4D default coordinates
+    call dist_appendFormat(dist_fmtX,        one,  1)
+    call dist_appendFormat(dist_fmtPX,       one,  2)
+    call dist_appendFormat(dist_fmtY,        one,  3)
+    call dist_appendFormat(dist_fmtPY,       one,  4)
+
+  case("6D") ! 6D default coordinates
+    call dist_appendFormat(dist_fmtX,        one,  1)
+    call dist_appendFormat(dist_fmtPX,       one,  2)
+    call dist_appendFormat(dist_fmtY,        one,  3)
+    call dist_appendFormat(dist_fmtPY,       one,  4)
+    call dist_appendFormat(dist_fmtZETA,     one,  5)
+    call dist_appendFormat(dist_fmtDPP0,     one,  6)
+
+  case("NORM") ! 6D normalised coordinates
+    call dist_appendFormat(dist_fmtXN,       one,  1)
+    call dist_appendFormat(dist_fmtPXN,      one,  2)
+    call dist_appendFormat(dist_fmtYN,       one,  3)
+    call dist_appendFormat(dist_fmtPYN,      one,  4)
+    call dist_appendFormat(dist_fmtZN,       one,  5)
+    call dist_appendFormat(dist_fmtPZN,      one,  6)
+
+  case("ACTION") ! 6D action
+    call dist_appendFormat(dist_fmtJX,       one,  1)
+    call dist_appendFormat(dist_fmtPhiX,     one,  2)
+    call dist_appendFormat(dist_fmtJY,       one,  3)
+    call dist_appendFormat(dist_fmtPhiY,     one,  4)
+    call dist_appendFormat(dist_fmtJZ,       one,  5)
+    call dist_appendFormat(dist_fmtPhiZ,     one,  6)
+
+  case("IONS") ! The ion columns
+    call dist_appendFormat(dist_fmtMASS,     c1e3, 0)
+    call dist_appendFormat(dist_fmtCHARGE,   one,  0)
+    call dist_appendFormat(dist_fmtIonA,     one,  0)
+    call dist_appendFormat(dist_fmtIonZ,     one,  0)
+    call dist_appendFormat(dist_fmtPDGID,    one,  0)
+
+  case("OLD_DIST") ! The old DIST block file format
+    call dist_appendFormat(dist_fmtPartID,   one,  0)
+    call dist_appendFormat(dist_fmtParentID, one,  0)
+    call dist_appendFormat(dist_fmtNONE,     one,  0)
+    call dist_appendFormat(dist_fmtX,        c1e3, 1)
+    call dist_appendFormat(dist_fmtY,        c1e3, 3)
+    call dist_appendFormat(dist_fmtNONE,     one,  0)
+    call dist_appendFormat(dist_fmtXP,       c1e3, 2)
+    call dist_appendFormat(dist_fmtYP,       c1e3, 4)
+    call dist_appendFormat(dist_fmtNONE,     one,  0)
+    call dist_appendFormat(dist_fmtIonA,     one,  0)
+    call dist_appendFormat(dist_fmtIonZ,     one,  0)
+    call dist_appendFormat(dist_fmtMASS,     c1e3, 0)
+    call dist_appendFormat(dist_fmtP,        c1e3, 6)
+    call dist_appendFormat(dist_fmtDT,       one,  5)
+
+  case default
+    write(lerr,"(a)") "DIST> ERROR Unknown column format '"//trim(fmtName)//"'"
+    fErr = .true.
+
+  end select
+
+end subroutine dist_setColumnFormat
+
+! ================================================================================================ !
+!  Parse File Column Units
```

Comment:
Maybe make the comment a bit more descriptive of what you do here, i.e. find the proper unit conversion

Reply:
Yeah, it's just out of date. Units weren't there at the start.